File: adventofcode-2016/src/main/java/nl/mvdr/adventofcode/adventofcode2016/day16/Part1.java
package nl.mvdr.adventofcode.adventofcode2016.day16;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import nl.mvdr.adventofcode.solver.Solver;

public class Part1 implements Solver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part1.class);

    @Override
    public String solve(String inputfile) {
        return new DragonChecksumSolver(272).solve(inputfile);
    }

    public static void main(String[] args) {
        Part1 instance = new Part1();

        String result = instance.solve("input-day16.txt");

        LOGGER.info(result);
    }
}

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day05/Part1.kt
package nl.mvdr.adventofcode.adventofcode2025.day05

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.adventofcode2025.day04.parseInput
import nl.mvdr.adventofcode.point.Point
import nl.mvdr.adventofcode.range.LongRange
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart1(linesSequence: Sequence<String>) = parseInput(linesSequence.toList()).countFreshIngredients()

fun main() {
    val result = FunctionSolver(::solvePart1).solve("input-day05.txt")
    logger.info { result }
}
